src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs

```diff
@@ -43,7 +43,7 @@ protected AbstractHealthCheck(TParameters parameters, ILogger logger, ActivitySo
 		/// <inheritdoc />
 		public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken token = default)
 		{
-			using Activity? activity = m_activitySource.StartActivity("HealthCheckActivity")
+			using Activity? activity = m_activitySource.StartActivity(context.Registration.Name)
```

Comment:
Suggest adding an IsNullOrWhiteSpace check on this value, falling back to "HealthCheckActivity" if the name is not set correctly.